tests/std/tests/P0896R4_ranges_alg_uninitialized_move/test.cpp

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <memory>
+#include <ranges>
+#include <span>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+#ifndef _CONSTEXPR20_DYNALLOC // TRANSITION, P0784
+#define _CONSTEXPR20_DYNALLOC
+#endif
+
+// Validate that uninitialized_move_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::uninitialized_move_result<int, double>, ranges::in_out_result<int, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_move(borrowed<false>{}, borrowed<false>{})),
+    ranges::uninitialized_move_result<ranges::dangling, ranges::dangling>>);
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_move(borrowed<false>{}, borrowed<true>{})),
+    ranges::uninitialized_move_result<ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_move(borrowed<true>{}, borrowed<false>{})),
+    ranges::uninitialized_move_result<int*, ranges::dangling>>);
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_move(borrowed<true>{}, borrowed<true>{})),
+    ranges::uninitialized_move_result<int*, int*>>);
+
+struct int_wrapper {
+    inline static int constructions = 0;
+    inline static int destructions  = 0;

```

Comment:
Static data members can be inline?

Reply:
Yes. `static constexpr` data members are implicitly `inline`, non-`constexpr` are not.